File: week6/pset6/sentimental-readability/readability.py
import string


def count_letters(text):
    num_letters = 0
    for c in text:
        if c in string.ascii_letters:
            num_letters += 1
    return num_letters


def count_words(text):
    n_s = text.split()
    return len(n_s)


def count_sentences(text):
    num_sentences = 0
    ends = ['.', '!', '?']
    for c in text:
        if c in ends:
            num_sentences += 1
    return num_sentences


def CLi(text):
    """calculate Coleman-Liau index"""
    W = count_words(text)
    aL = count_letters(text)
    aS = count_sentences(text)
    L = (aL / W) * 100
    S = (aS / W) * 100
    return (0.0588 * L - 0.296 * S - 15.8)


text = input("Text: ")
Grade = CLi(text)


if Grade < 1:
    print("Before Grade 1")
elif Grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(Grade)}")

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = (letters / (float) words) * 100.0;
    float S = (sentences / (float) words) * 100.0;
    int index = (int) round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int n = strlen(text);
    int nums_letters = 0;

    for (int i = 0; i < n; i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            nums_letters += 1;
        }
        else if (text[i] >= 'A' && text[i] <= 'Z')
        {
            nums_letters += 1;
        }
    }
    return nums_letters;
}

int count_words(string text)
{
    int n = strlen(text);
    int nums_words = 0;

    for (int i = 0; i < n; i++)
    {
        if (text[i] == ' ')
        {
            nums_words += 1;
        }
    }
    nums_words += 1;
    return nums_words;
}

int count_sentences(string text)
{
    int n = strlen(text);
    int nums_sentences = 0;

    for (int i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            nums_sentences += 1;
        }
    }
    return nums_sentences;
}